### DISCUSSION TITLE
Release/5.3.0-beta0

### DISCUSSION THREAD

**Discussion by Athanasius** [EDCD/EDMarketConnector#1365] [Announcement]
This pre-release is intended for testing building under Python 3.10(.0).

There is one other small change:

* We now set a custom User-Agent on all remote-server queries we make.  This is of the form `EDCD-EDMarketConnector-<full version>`, i.e. `EDCD-EDMarketConnector-5.3.0-beta0+d546e840.DIRTY`.  This will aid in debugging any issues EDMC might have with services like EDDN, EDDB and the like.

This should have no direct impact on user experience.

Primarily we just need to ensure that this version installs, runs and doesn't exhibit any weirdness **that isn't also present in 5.2.3**.

<hr /><em>This discussion was created from the release <a href='https://github.com/EDCD/EDMarketConnector/releases/tag/Release/5.3.0-beta0'>Release/5.3.0-beta0</a>.</em>

**Comment by Athanasius** [comment]
If anyone has tried this out and found it then doesn't even run, then I just figured out why.

TL;DR - Python.org has older versions of some files in 3.10.0 than in the, released later, 3.9.9, so the windows installer thinks the 3.10.0 versions of those files shouldn't be installed, but still removes the newer ones from the prior EDMC version anyway.  Workaround is to either:

1. Remove EDMarketConnector before installing 5.3.0-beta0.
    1. Open Windows Settings (`WinKey`+`I`).
    2. Select `Apps`.
    3. Type `Market` in the search box.
    4. Click the entry for `Elite Dangerous Market Connector`.
    5. Click the `Uninstall` button, and then the additional `Uninstall` button in the pop-up.
    6. Now run the 5.3.0-beta0 installer again.
2. OR Perform a 'Repair' install after attempting the Upgrade.
    1. Find where you downloaded `EDMarketConnector_win_5.3.0-beta0.msi` to
    2. Right click the file.
    3. Select the 'Repair' option.
    NB: The program won't automatically run after the Repair


Long Version
===
Python 3.10.0 was released on 2021-10-04, when the latest 3.9.x version was 3.9.7 (released 2021-08-30).  This has version 8.6.2.10 of the two files `tcl86t.dll` and `tk86t.dll`.  Those files are part of the Tcl/Tk system, as used by python's `tkinter` module.  That's responsible for all of the EDMC UI.

Python 3.9.9, released 2021-11-15, and used by us in EDMC 5.2.3, had the version 8.6.2.12 of the two DLL files, so newer.

Thus when we build an installer using Python 3.10.0 it includes **older versions of the two DLL files** and, for reasons still unknown, Windows decides to both uninstall the 'old install, but newer version' of the files and then not install the 'new install, but older version' of each file.  This might be related to some parts of our EDMarketConnector.wxs file aimed at always doing a full reinstall, rather than patching, but that still doesn't explain exactly what happens here.

Thus an install of 5.3.0-beta0 made over 5.2.3 ends up without `tcl86t.dll` or `tk86t.dll` and will cause an error relating to importing the file `_tkinter.pyd` (a compiled version of the tkinter module files) and the EDMC program will error out before any UI is visible.

So, we'll be waiting for Python.org to release 3.10.1 (apparently due on 2021-12-06, so tomorrow as I type this), or later, that includes at least the same version of the files as Python 3.9.9 does.
